python-scripts/luogu_spider.py

```
@@ -9,67 +9,92 @@ class LuoguFriendLinkSpider:
+                    response.status_code in [403, 429] or
+                    '访问过于频繁' in response.text or
+                    'too frequent' in response.text.lower()
+                )
+
+                if is_rate_limited:
+                    if response.status_code == 403 and '访问过于频繁' not in response.text:
+                        print("403 Forbidden without rate-limit hint. Cookies may be invalid.")
+                        print(f"Response preview: {response.text[:200]}")
+                        break
+
+                    wait_time = (2 ** retry_count) + random.uniform(2, 6)
```

Ruff (0.16.1)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)